src/io.md

```diff
@@ -76,3 +76,35 @@ that error explicit.
 
 Note that buffering also works with stdout, so you might want to combine manual
 locking *and* buffering when making many writes to stdout.
+
+## Reading Input as Raw Bytes
+
+The built-in [String] type uses UTF-8 internally, which adds a small, but
+nonzero overhead caused by UTF-8 validation when you read input into it. If you
+just want to process input bytes without worrying about UTF-8 (for example if
+you handle ASCII text), you can use [`BufRead::read_until`]:
+```rust
+use std::io::{BufReader, BufRead};
+
+# fn blah() -> Result<(), std::io::Error> {
+let stdin = std::io::stdin();
+let mut stdin = stdin.lock();
+let mut buffered = BufReader::new(stdin);
+let mut buf = vec![];
+
+loop {
```

Comment:
Hmm, using `loop` is bogus, right? This will just infinitely loop. This example isn't adding much, I think it can just be removed. (Sorry I didn't notice this earlier.)

Reply:
The example is not complete, you are right. Probably it's not needed.